lib/jxl/enc_ac_strategy.cc

```diff
@@ -1016,20 +1095,52 @@ void ProcessRectACSNew(PassesEncoderState* JXL_RESTRICT enc_state,
 
   // Priority is a tricky kludge to avoid collisions so that transforms
   // don't overlap.
-  uint8_t priority[64] = {0};
+  uint8_t priority[64] = {};
   for (auto tx : kTransformsForMerge) {
     AcStrategy acs = AcStrategy::FromRawStrategy(tx.type);
     for (size_t cy = 0; cy + acs.covered_blocks_y() - 1 < rect.ysize();
          cy += acs.covered_blocks_y()) {
       for (size_t cx = 0; cx + acs.covered_blocks_x() - 1 < rect.xsize();
            cx += acs.covered_blocks_x()) {
-        TryMergeToACSCandidate(tx.type, bx, by, cx, cy, config, cmap_factors,
-                               ac_strategy, tx.entropy_mul, tx.priority,
-                               &priority[0], entropy_estimate, block,
-                               scratch_space, quantized);
+        if (cy + 1 < rect.ysize() && cx + 1 < rect.xsize()) {
+          if (tx.type == AcStrategy::Type::DCT8X16) {
+            // We handle both DCT8X16 and DCT16X8 at the same time.
+            if ((cy | cx) % 2 == 0) {
+              Try16X8And8X16(bx, by, cx, cy, config, cmap_factors, ac_strategy,
+                             tx.entropy_mul, entropy_estimate, block,
+                             scratch_space, quantized);
+            }
+            continue;
+          } else if (tx.type == AcStrategy::Type::DCT16X8) {
+            // We handled both DCT8X16 and DCT16X8 at the same time,
+            // and that is above. The last column and last row,
+            // when the last column or last row is odd numbered,
+            // are still handled by TryMergeAcs.
+            continue;
+          }
+        }
+        if ((tx.type == AcStrategy::Type::DCT8X16 && cy % 2 == 1) ||
+            (tx.type == AcStrategy::Type::DCT16X8 && cx % 2 == 1)) {
```

Comment:
Aren't `cx` and `cy` swapped here?

Reply:
No, it is correct.

16x8 means y=16 and x=8, we made that in a bizarre way.

This means that 16x8 it is a vertical block.

If a vertical block is on an odd numbered column, it has already been processed. Only necessary to merge it in the generic process is there are an odd number of columns, i.e., cx % 2 == 0.